go-controller/pkg/cni/helper_linux.go

```diff
@@ -818,3 +843,48 @@ func (pr *PodRequest) deletePorts(ifaces []string, podNamespace, podName string)
 		pr.deletePort(iface, podNamespace, podName)
 	}
 }
+
+func getGWLinkLocalAddress(namespace, podName, nadName string, getter PodInfoGetter) (string, error) {
+	pod, err := getter.getPod(namespace, podName)
+	if err != nil {
+		return "", err
+	}
+
+	podNetworks, err := util.UnmarshalPodAnnotationAllNetworks(pod.Annotations)
+	if err != nil {
+		return "", err
+	}
+
+	return podNetworks[nadName].IPv6LLAGateway, err
```

Comment:
are we sure we'll always find `podNetworks[nadName]` ?

Reply:
On non interconnect we won't have it due to the [bug](https://github.com/ovn-kubernetes/ovn-kubernetes/issues/4876), Once that bug is fixed, we will be able to assert here in case desired.
The previous PR adds it on IC [only](https://github.com/ovn-kubernetes/ovn-kubernetes/pull/4878/files#diff-3d3fe81a31268662d021e60eb37f4a1f7e96ddb7e61de7f49acbbd2ae82e81caR595).

Anyhow, if it is not in the map (and assuming we are on IC), it is an higher level bug because it shouldn't happen,
in this PR we just don't config the nftables in such [case](https://github.com/ovn-kubernetes/ovn-kubernetes/pull/4852/files#diff-01a2ef868473804cd3d9b894b093e5be0735cfb21bf27f375ece3a33cc07edc9R256)

(p.s we just added a variable to the existing annotation)